### PR TITLE
docs: Make docs badge link to latest stable version on ReadTheDocs

### LINKED ISSUE #1286: Make default docs link point to latest stable release on ReadTheDocs
# Description

We have enough users and enough versions out that we should prioritize having the docs be versioned cleanly along with the releases. At the moment our docs links point to the docs build from `master` but we should probably have the default view of the docs correspond to the latest stable release on PyPI &mdash; so at the time of writing that would be the `v0.5.4` API.

We already have versioned docs on ReadTheDocs, so that might just mean adding a link to the README of https://pyhf.readthedocs.io/

[RTD_docs_versioned](https://pyhf.readthedocs.io/)

(Note: This has the [`v0.6.0`](https://github.com/scikit-hep/pyhf/projects/10) label so that this will go in and get picked up for the upcoming PyPI release.)

### PR DESCRIPTION
**Pull request by matthewfeickert**
# Description

Resolves #1286 

Make the default badge on the `README` link to the docs for the latest stable release of `pyhf` hosted on ReadTheDocs.

Example for `v0.5.4` (latest at the time of PR):

[RTD latest stable release](https://pyhf.readthedocs.io/)

```markdown
[![RTD latest stable release](https://img.shields.io/badge/docs-latest-blue.svg)](https://pyhf.readthedocs.io/)
```

ReadTheDocs build: https://pyhf.readthedocs.io/en/docs-make-default-docs-rtd/

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Add badge to README for latest versioned docs on ReadTheDocs
   - version number in badge display URL is updated through bump2version
* Change PyPI documentation URL to also point to ReadTheDocs
* Remove LGTM status badge as service is no longer used
```

### CODE REVIEW

**Comment by matthewfeickert** [comment]
This should probably also deal with the citation as well. Hm.

https://github.com/scikit-hep/pyhf/blob/b0283eedcc603c963fd991edd73b9d4e18273628/src/pyhf/data/citation.bib#L6

File: setup.cfg [hunk 1]
```diff
@@ -11,7 +11,7 @@ license = Apache
 license_file = LICENSE
 keywords = physics fitting numpy scipy tensorflow pytorch jax
 project_urls =
-    Documentation = https://scikit-hep.org/pyhf/
+    Documentation = https://pyhf.readthedocs.io/en/v0.5.4/
```
Comment (matthewfeickert) [inline setup.cfg:14]: Not sure if this is a good idea or not to change _everywhere_, but feels like at least for PyPI and GitHub might be a good idea.

**Comment by matthewfeickert** [comment]
@lukasheinrich @kratsg  If you run

```
$ git grep "scikit-hep.org/pyhf" | grep --invert-match "schema" | grep scikit-hep
CONTRIBUTING.md:If you have suggestions, questions that you can't find answers to on the [documentation website](https://scikit-hep.org/pyhf/) or on the [GitHub Discussions](https://github.com/scikit-hep/pyhf/discussions), or have found a bug please [open an Issue](https://github.com/scikit-hep/pyhf/issues/new/choose)!
CONTRIBUTING.md:We recommend first reading the "[Developing](https://scikit-hep.org/pyhf/development.html)" page on the pyhf website and the coming back here.
README.rst:Citations <https://scikit-hep.org/pyhf/citations.html>`__, the preferred
docs/examples/notebooks/toys.ipynb:    "You'll notice that the shape for `mc_bkg` and `mc_sig` is not the input shape we passed in `(10000,)` but rather `(10000,2)`! Why is that? The HistFactory model is a product of many separate pdfs: Poissons representing the main model, and Gaussians representing the auxiliary measurements. In `pyhf`, this is represented under the hood as a [`Simultaneous`](https://scikit-hep.org/pyhf/_generated/pyhf.probability.Simultaneous.html) pdf of the main model and the auxiliary model &mdash; hence the second dimension.\n",
```

I think it is is okay to leave https://scikit-hep.org/pyhf/ in `CONTRIBUTING.md` and in the citations section for `README.rst` as:
- If someone is reading `CONTRIBUTING.md` they should be dealing with the dev docs anyway.
- We always want the latest citation out there

Thoughts?

File: setup.cfg [hunk 2]
```diff
@@ -11,7 +11,7 @@ license = Apache
 license_file = LICENSE
 keywords = physics fitting numpy scipy tensorflow pytorch jax
 project_urls =
-    Documentation = https://scikit-hep.org/pyhf/
+    Documentation = https://pyhf.readthedocs.io/
```
Comment (matthewfeickert) [inline setup.cfg:14]: Thoughts on changing it here too? Might not be a bad idea to direct people from PyPI to stable docs.

Comment (kratsg) [reply setup.cfg:14]: this probably makes the most sense.

**Comment by kratsg** [comment]
> * If someone is reading `CONTRIBUTING.md` they should be dealing with the dev docs anyway.
> * We always want the latest citation out there

Yeah, I think both make sense.

**Review by kratsg — APPROVED** [review]

**Pull request merged**